micrometer-core/src/test/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsCommandListenerTest.java

```diff
@@ -206,7 +207,27 @@ void shouldSupportConcurrentCommands() throws InterruptedException {
 
     @AfterEach
     void destroy() {
-        mongo.close();
+        if (mongo != null) {
+            mongo.close();
+        }
+    }
+
+    private MongoClient createMongoClient(AtomicReference<String> clusterId) {
+        // tag::setup[]
+        MongoClientSettings settings = MongoClientSettings.builder()
+            .addCommandListener(new MongoMetricsCommandListener(registry))
+            .applyToClusterSettings(builder -> builder.hosts(singletonList(new ServerAddress(host, port)))
+                .addClusterListener(new ClusterListener() {
+                    @Override
+                    public void clusterOpening(ClusterOpeningEvent event) {
+                        clusterId.set(event.getClusterId().getValue());
```

Comment:
I think we don't want this to end up in the documentation snippet, but since it was there before, this isn't changing anything. We can tackle getting that out of the documentation in another PR.

Reply:
I tried removing it from the docs by adding an extra tag `// tag::setup_2[] `, but I couldn't get the docs to build locally and didn't want to risk breaking the indentation.

If we end up creating a separate issue for the docs, it might be worth adding something like a docker-compose to make local doc building easier. (if it works - I'm not familiar with antora). - what do you think?

for eg, Testcontainers has a setup like this for mkdocs: 
https://github.com/testcontainers/testcontainers-java/blob/main/docker-compose.yml